server/api/availability.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package api
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/robfig/cron/v3"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/job"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/remote"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils/bot"
+)
+
+const (
+	AVAILABILITY_VIEW_FREE              = '0'
+	AVAILABILITY_VIEW_TENTATIVE         = '1'
+	AVAILABILITY_VIEW_BUSY              = '2'
+	AVAILABILITY_VIEW_OUT_OF_OFFICE     = '3'
+	AVAILABILITY_VIEW_WORKING_ELSEWHERE = '4'
+)
+
+type availabilityJob struct {
+	api API
+}
+
+func NewAvailabilityJob(api API) job.RecurringJob {
+	return &availabilityJob{api: api}
+}
+
+func (j *availabilityJob) Run() {
+	c := cron.New()
+	c.AddFunc("* * * * *", j.Work)
+	c.Start()
+}
+
+func (j *availabilityJob) getLogger() bot.Logger {
+	return j.api.(*api).Logger
+}
+
+func (j *availabilityJob) Work() {
+	log := j.getLogger()
+	log.Debugf("Availability job beginning")
+
+	_, err := j.api.GetUserAvailability()
+	if err != nil {
+		log.Errorf("Error during Availability job", "error", err.Error())
+	}
+
+	log.Debugf("Availability job finished")
+}
+
+func (api *api) GetUserAvailability() (string, error) {
```

Comment:
This is called by a single user using `/availability`, possibly just for debugging?

Reply:
Suggest a different name, since this actually _Sets_ the user status.

2/5 `api.SyncStatusForSingleUser` and `api.SyncStatusForUsers`. I am not really concerned about long names for high-level `api` methods since they tend to be called rarely, once/user action. Being intention-revealing is important for those.

0/5 I'd consider restructuring the code around (a) getting the desired status, and (b) setting it, for 1 and a collection of users (whether it's "all" or a subset should be up to the caller).